crates/edr_provider/src/error.rs

```diff
@@ -164,59 +164,73 @@ pub enum ProviderError {
-            ProviderError::TransactionFailed(transaction_failure) => (
-                -32000,
-                Some(
-                    serde_json::to_value(transaction_failure).expect("transaction_failure to json"),
-                ),
+        let code = match &value {
+            ProviderError::AccountOverrideConversionError(_) => -32000,
+            ProviderError::AutoMineGasPriceTooLow { .. } => -32000,
+            ProviderError::AutoMineMaxFeeTooLow { .. } => -32000,
+            ProviderError::AutoMineNonceTooHigh { .. } => -32000,
+            ProviderError::AutoMineNonceTooLow { .. } => -32000,
+            ProviderError::AutoMinePriorityFeeTooLow { .. } => -32000,
+            ProviderError::Blockchain(_) => -32000,
+            ProviderError::Creation(_) => -32000,
+            ProviderError::InvalidArgument(_) => -32602,
+            ProviderError::InvalidBlockNumberOrHash { .. } => -32000,
+            ProviderError::InvalidBlockTag { .. } => -32602,
+            ProviderError::InvalidChainId { .. } => -32602,
+            ProviderError::InvalidDropTransactionHash(_) => -32602,
+            ProviderError::InvalidFilterSubscriptionType { .. } => -32602,
+            ProviderError::InvalidInput(_) => -32000,
+            ProviderError::InvalidTransactionIndex(_) => -32602,
+            ProviderError::InvalidTransactionInput(_) => -32000,
+            ProviderError::InvalidTransactionType(_) => -32602,
+            ProviderError::MemPoolUpdate(_) => -32000,
+            ProviderError::MineBlock(_) => -32000,
+            ProviderError::MinerTransactionError(_) => -32000,
+            ProviderError::RpcVersion(_) => -32000,
+            ProviderError::RunTransaction(_) => -32000,
+            ProviderError::Serialization(_) => -32000,
+            ProviderError::SetAccountNonceLowerThanCurrent { .. } => -32000,
+            ProviderError::SetAccountNonceWithPendingTransactions => -32603,
+            ProviderError::SetMinGasPriceUnsupported => -32000,
+            ProviderError::SetNextBlockBaseFeePerGasUnsupported { .. } => -32000,
+            ProviderError::SetNextPrevRandaoUnsupported { .. } => -32000,
+            ProviderError::Signature(_) => -32000,
+            ProviderError::State(_) => -32000,
+            ProviderError::SystemTime(_) => -32000,
+            ProviderError::TimestampLowerThanPrevious { .. } => -32000,
+            ProviderError::TimestampEqualsPrevious { .. } => -32000,
+            ProviderError::TransactionFailed(_) => -32000,
+            ProviderError::TransactionCreationError(_) => -32000,
+            ProviderError::TryFromIntError(_) => -32000,
+            ProviderError::Unimplemented(_) => -32000,
+            ProviderError::UnknownAddress { .. } => -32000,
+            ProviderError::UnmetHardfork { .. } => -32602,
+            ProviderError::UnsupportedAccessListParameter { .. } => -32602,
+            ProviderError::UnsupportedEIP1559Parameters { .. } => -32602,
+        };
```

Comment:
Didn't change any numbers, just factored out the `data` as it's only used in one variant.